### PR TITLE
version 2.18.1

### PR DESCRIPTION
**Pull request by larskuhtz**
* [x] update cabal version to 2.18.1
* [x] update CHANGELOG.md
* [x] update freeze file for release build

### CODE REVIEW

**Comment by larskuhtz** [comment]
Dependency updates:

```diff
diff --git a/cabal.project.freeze b/cabal.project.freeze
index 3ab313d5..68019b58 100644
--- a/cabal.project.freeze
+++ b/cabal.project.freeze
@@ -40,16 +40,16 @@ constraints: any.Boolean ==0.2.4,
              any.base ==4.14.3.0,
              any.base-compat ==0.12.2,
              any.base-compat-batteries ==0.12.2,
-             any.base-orphans ==0.8.7,
+             any.base-orphans ==0.8.8.1,
              any.base-unicode-symbols ==0.2.4.2,
              base-unicode-symbols +base-4-8 -old-base,
              any.base16-bytestring ==0.1.1.7,
              any.base64-bytestring ==1.0.0.3,
              any.basement ==0.0.15,
-             any.bifunctors ==5.5.14,
+             any.bifunctors ==5.5.15,
              bifunctors +semigroups +tagged,
              any.binary ==0.8.8.0,
-             any.binary-orphans ==1.0.3,
+             any.binary-orphans ==1.0.4.1,
              any.bitvec ==1.1.3.0,
              bitvec -libgmp,
              any.blaze-builder ==0.4.2.2,
@@ -101,6 +101,7 @@ constraints: any.Boolean ==0.2.4,
              cryptonite -check_alignment +integer-gmp -old_toolchain_inliner +support_aesni +support_deepseq +support_pclmuldq +support_rdrand -support_sse +use_target_attributes,
              any.cuckoo ==0.3.1,
              cuckoo -mwc-random -pcg-random,
+             any.data-array-byte ==0.1.0.1,
              any.data-bword ==0.1.0.2,
              any.data-default ==0.7.1.1,
              any.data-default-class ==0.1.2.0,
@@ -113,10 +114,10 @@ constraints: any.Boolean ==0.2.4,
              any.dec ==0.0.5,
              any.deepseq ==1.4.4.0,
              any.dense-linear-algebra ==0.1.0.0,
-             any.deriving-compat ==0.6.2,
+             any.deriving-compat ==0.6.3,
              deriving-compat +base-4-9 +new-functor-classes +template-haskell-2-11,
              any.digraph ==0.3.0,
-             any.direct-sqlite ==2.3.27,
+             any.direct-sqlite ==2.3.28,
              direct-sqlite +fulltextsearch +haveusleep +json1 -systemlib +urifilenames,
              any.directory ==1.3.6.0,
              any.distributive ==0.6.2.1,
@@ -161,11 +162,11 @@ constraints: any.Boolean ==0.2.4,
              http2 -devel -doc -h2spec,
              any.indexed-list-literals ==0.2.1.3,
              any.indexed-traversable ==0.1.2,
-             any.indexed-traversable-instances ==0.1.1.1,
+             any.indexed-traversable-instances ==0.1.1.2,
              any.integer-gmp ==1.0.3.0,
              any.integer-logarithms ==1.0.3.1,
              integer-logarithms -check-bounds +integer-gmp,
-             any.invariant ==0.6,
+             any.invariant ==0.6.1,
              any.iproute ==1.7.12,
              any.ixset-typed ==0.5,
              any.js-chart ==2.9.4.1,
@@ -200,7 +201,7 @@ constraints: any.Boolean ==0.2.4,
              mtl-compat -two-point-one -two-point-two,
              any.mwc-probability ==2.3.1,
              any.mwc-random ==0.15.0.2,
-             any.network ==3.1.2.7,
+             any.network ==3.1.2.8,
              network -devel,
              any.network-byte-order ==0.1.6,
              any.network-uri ==2.6.4.2,
@@ -233,12 +234,12 @@ constraints: any.Boolean ==0.2.4,
              any.profunctors ==5.6.2,
              any.psqueues ==0.2.7.3,
              any.pvar ==1.0.0.0,
-             any.quickcheck-instances ==0.3.28,
+             any.quickcheck-instances ==0.3.29.1,
              quickcheck-instances -bytestring-builder,
              any.random ==1.2.1.1,
              any.recv ==0.1.0,
              any.reducers ==3.12.4,
-             any.reflection ==2.1.6,
+             any.reflection ==2.1.7,
              reflection -slow +template-haskell,
              any.resource-pool ==0.4.0.0,
              any.resourcet ==1.2.6,
@@ -289,7 +290,7 @@ constraints: any.Boolean ==0.2.4,
              any.strict-concurrency ==0.2.4.3,
              any.string-conversions ==0.4.0.1,
              any.syb ==0.7.2.2,
-             any.tagged ==0.8.6.1,
+             any.tagged ==0.8.7,
              tagged +deepseq +transformers,
              any.tasty ==1.4.3,
              tasty +unix,
@@ -323,12 +324,12 @@ constraints: any.Boolean ==0.2.4,
              transformers-compat -five +five-three -four +generic-deriving +mtl -three -two,
              any.trifecta ==2.1.1,
              any.type-equality ==1,
-             any.typed-process ==0.2.10.1,
+             any.typed-process ==0.2.11.0,
              any.unix ==2.7.2.2,
              any.unix-compat ==0.6,
              unix-compat -old-time,
              any.unix-time ==0.4.8,
-             any.unliftio ==0.2.23.0,
+             any.unliftio ==0.2.24.0,
              any.unliftio-core ==0.2.1.0,
              any.unordered-containers ==0.2.15.0,
              unordered-containers -debug,
@@ -363,10 +364,10 @@ constraints: any.Boolean ==0.2.4,
              any.x509-store ==1.6.9,
              any.x509-system ==1.6.7,
              any.x509-validation ==1.6.12,
-             any.yaml ==0.11.10.0,
+             any.yaml ==0.11.11.0,
              yaml +no-examples +no-exe,
              any.yet-another-logger ==0.4.1,
              yet-another-logger -tbmqueue,
              any.zlib ==0.6.3.0,
              zlib -bundled-c-zlib -non-blocking-ffi -pkg-config
-index-state: hackage.haskell.org 2023-02-14T18:10:13Z
+index-state: hackage.haskell.org 2023-03-06T16:51:46Z
```

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,31 @@
...
+
+Performance Improvements:
+
+*   Optimize JSON+base64 encoding. (#1611)
+*   Use `application/octet-stream` encoding for P2P header queries. (#1619)
+
+Miscellaneous:
+
+*   Remove unused rate limiting configuration settings. (#1616)
+*   Remove CORS support from P2P AP. (#1616)
```
Comment (jwiegley) [inline CHANGELOG.md:22]:
```suggestion
*   Remove CORS support from the P2P API. (#1616)
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,31 @@
...
+*   Remove unsused hashes and SPV endpoints from P2P API. (#1616)
```
Comment (jwiegley) [inline CHANGELOG.md:23]:
```suggestion
*   Remove unused hashes and SPV endpoints from the P2P API. (#1616)
```

**Review by jwiegley — APPROVED** [review]

**Review by edmundnoble — APPROVED** [review]
Noting that this updates our version of sqlite.

**Pull request merged**